tests/pytests/functional/states/test_gpg.py

```diff
@@ -0,0 +1,150 @@
+import shutil
+import subprocess
+
+import psutil
+import pytest
+
+try:
+    import gnupg as gnupglib
+
+    HAS_GNUPG = True
+except ImportError:
+    HAS_GNUPG = False
+
+
+pytestmark = [
+    pytest.mark.skipif(HAS_GNUPG is False, reason="Needs python-gnupg library"),
+    pytest.mark.skip_if_binaries_missing("gpg", reason="Needs gpg binary"),
+]
+
+
+@pytest.fixture
+def gpghome(tmp_path):
+    root = tmp_path / "gpghome"
+    root.mkdir(mode=0o0700)
+    try:
+        yield root
+    finally:
+        # Make sure we don't leave any gpg-agents running behind
+        gpg_connect_agent = shutil.which("gpg-connect-agent")
+        if gpg_connect_agent:
+            gnupghome = root / ".gnupg"
+            if not gnupghome.is_dir():
+                gnupghome = root
+            try:
+                subprocess.run(
+                    [gpg_connect_agent, "killagent", "/bye"],
+                    env={"GNUPGHOME": str(gnupghome)},
+                    shell=False,
+                    check=True,
+                    stdout=subprocess.DEVNULL,
+                    stderr=subprocess.DEVNULL,
+                )
+            except subprocess.CalledProcessError:
+                # This is likely CentOS 7 or Amazon Linux 2
+                pass
+
+        # If the above errored or was not enough, as a last resort, let's check
+        # the running processes.
+        for proc in psutil.process_iter():
+            try:
+                if "gpg-agent" in proc.name():
+                    for arg in proc.cmdline():
+                        if str(root) in arg:
+                            proc.terminate()
+            except Exception:  # pylint: disable=broad-except
+                pass
```

Comment:
I honestly just copied this from the preexisting execution module unit tests. Incidentally it seems you wrote that code: https://github.com/saltstack/salt/commit/b5912dd349a3b848bd38edda03958bf09867e919

If I had to guess, it might be finicky to ensure the gpg-agent is actually dead.

Reply:
> I honestly just copied this from the preexisting execution module unit tests. Incidentally it seems you wrote that code: [b5912dd](https://github.com/saltstack/salt/commit/b5912dd349a3b848bd38edda03958bf09867e919)
> 
> If I had to guess, it might be finicky to ensure the gpg-agent is actually dead.

Heh, ok. Guess there was a reason back then.